js/widgets/page.js

```diff
@@ -73,36 +73,15 @@ $.widget( "mobile.page", {
 		if ( this.options.enhanceWithin ) {
 			this.element.enhanceWithin();
 		}
-		// Dialog widget is deprecated in 1.4 remove this in 1.5
-		if ( $.mobile.getAttribute( this.element[ 0 ], "role" ) === "dialog" && $.mobile.dialog ) {
-			this.element.dialog();
-		}
 	},
 
 	_enhance: function() {
-		var attrPrefix = "data-" + $.mobile.ns,
-			self = this;
-
 		if ( this.options.role ) {
```

Comment:
Unfortunately, it's not that simple. In the case of the page, you're right, it's merely short-circuit DOM manip, but in the case of the dialog extension, it wraps the content, so in the non-pre-rendered case of the dialog extension, it makes sense to add the classes while the contents is detached. I need to think about this a bit.